From: SeongJae Park <sj@kernel.org>
Date: Mon, 25 Aug 2025 07:46:24 -0700
Subject: [PATCH] mm/damon/paddr: use do_div() on i386 for damon_pa_migrate()
 return value

Otherwise, __udivdi3 linking issue happens on certain configs.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 6fb92ecc513d..34248b742a73 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -286,7 +286,7 @@ static unsigned long damon_pa_migrate(struct damon_region *r,
 	applied = damon_migrate_pages(&folio_list, s->target_nid);
 	cond_resched();
 	s->last_applied = folio;
-	return applied * PAGE_SIZE / addr_unit;
+	return damon_pa_core_addr(applied * PAGE_SIZE, addr_unit);
 }
 
 static unsigned long damon_pa_stat(struct damon_region *r,
-- 
2.39.5

